### DISCUSSION TITLE
Can you please explain angle and weight calculations a bit?

### DISCUSSION THREAD

**Discussion by Stanko** [PavelLaptev/warp-svg#8] [Q&A] [answered]
Hey there,
Thank you for the cool showcase of SVG warping!

I thought that idea of getting the initial weights, for each SVG point, and then using them to rebuild the position when control points are changed is really nifty. But I'll be honest - I don't really understand all of the math used. 

It would be super helpful if you have time to explain these two code segments:

Calculating angle:
https://github.com/PavelLaptev/warp-svg/blob/adc34885ed190df517199b7c724729fa6333a085/src/js/index.js#L116-L119

Calculating weight:
https://github.com/PavelLaptev/warp-svg/blob/adc34885ed190df517199b7c724729fa6333a085/src/js/index.js#L129-L131

Cheers!

**Comment by PavelLaptev** [comment, marked as answer]
Hi @Stanko! Thank you for feedback :-)
Actually, I also want to be honest — have no idea, because I copy/paste the code from this discussion https://github.com/benjamminf/warpjs/issues/11#issuecomment-313276116

And the library I used — https://github.com/benjamminf/warpjs

**Reply by Stanko** [reply]
Hehe thanks! I see author used "barycentric coordinates" to achieve this, so I'll do some research on that.
Cheers!
